### PR TITLE
GDScript: Add some checks for `@tool` and `@icon`

### LINKED ISSUE #76059: No error if a script has multiple `@tool` annotations at the top.
### Godot version

4.0.2

### System information

All Systems

### Issue description

Title is explanatory enough.

```
@tool
@tool
extends Node
```

I expect an error or warning on line 2, because it doesn't do anything.

### Steps to reproduce

N/A

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by dalexeev**
Closes #76059.

### CODE REVIEW

**Comment by YuriSizov** [comment]
Thanks! Btw, why err if a non-global class has an icon? We can still use them in the editor. I even has a PR adding support for those.

**Comment by dalexeev** [comment]
> I even has a PR adding support for those.

At least it doesn't currently work. This is even written in the docs, but there is no check in the parser.

> Add a custom icon to the current script. The script must be registered as a global class using the class_name keyword for this to have a visible effect.

> Btw, why err if a non-global class has an icon?

Probably because the editor reads this data from the `.godot/global_script_class_cache.cfg` file that only contains global named classes.

**Comment by YuriSizov** [comment]
Yes, in my PR I add a way to get it directly from the script.

**Comment by dalexeev** [comment]
> Yes, in my PR I add a way to get it directly from the script.

Okay, then I will remove that part. I will not delete the obsolete note in the docs, please do it within #75656.

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -3686,6 +3690,14 @@ bool GDScriptParser::icon_annotation(const AnnotationNode *p_annotation, Node *p
...
 	ClassNode *p_class = static_cast<ClassNode *>(p_node);
+	if (!p_class->icon_path.is_empty()) {
+		push_error(R"("@icon" annotation can only be used once.)", p_annotation);
+		return false;
+	}
+	if (String(p_annotation->resolved_arguments[0]).is_empty()) {
+		push_error(R"("@icon" annotation argument must contain the path to the icon.)", p_annotation->arguments[0]);
+		return false;
+	}
 	p_class->icon_path = p_annotation->resolved_arguments[0];
```
Comment (aaronfranke) [inline modules/gdscript/gdscript_parser.cpp:3701]: Considering this check isn't necessary at runtime, I think we should surround these with `#ifdef DEBUG_ENABLED`.

Comment (dalexeev) [reply modules/gdscript/gdscript_parser.cpp]: Done.

File: modules/gdscript/gdscript_parser.cpp [hunk 2]
```diff
@@ -3686,6 +3686,12 @@ bool GDScriptParser::validate_annotation_arguments(AnnotationNode *p_annotation)
 }
 
 bool GDScriptParser::tool_annotation(const AnnotationNode *p_annotation, Node *p_node) {
+#ifdef DEBUG_ENABLED
+	if (this->_is_tool) {
```
Comment (aaronfranke) [inline modules/gdscript/gdscript_parser.cpp:3690]: `this->` is redundant but it's consistent with the below, so I guess it's OK.

**Review by aaronfranke — APPROVED** [review]
I think this check is a good idea, even though it is fixing a very minor problem.

**Review by vnen — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**